src/server/pfs/fuse/server.go

```diff
@@ -858,6 +872,24 @@ func jsonMarshal(t interface{}) ([]byte, error) {
 	return buffer.Bytes(), err
 }
 
+func hasRepoRead(permissions []auth.Permission) bool {
+	for _, p := range permissions {
+		if p == auth.Permission_REPO_READ {
+			return true
+		}
+	}
+	return false
+}
+
+func hasRepoWrite(permissions []auth.Permission) bool {
```

Comment:
We don't need to check for the role `repoOwner` because we are checking for the specific write permission inside of the role? Would we have to add a similar function in the future if we wanted to check for other privileges like:
```
auth.Permission_REPO_DELETE,
auth.Permission_REPO_INSPECT_COMMIT,
auth.Permission_REPO_LIST_COMMIT,
auth.Permission_REPO_DELETE_COMMIT,
auth.Permission_REPO_CREATE_BRANCH,
auth.Permission_REPO_LIST_BRANCH,
auth.Permission_REPO_DELETE_BRANCH,
auth.Permission_REPO_LIST_FILE
```

Reply:
Good question. A user (i.e. root, robot, pipeline, etc.) can't be created by mixing and matching permissions. They can only be given roles (i.e. repoReader, repoWriter, etc.). https://pachyderm.slack.com/archives/C02C8TUF493/p1653398328191089

As a result, the roles that contain the `repo_write` permission are finite and known: `repoWriter`, `repoOwner`, and `clusterAdmin`. We're okay with these roles writing to a repo which means checking to see if a user has the `repo_write` permission is sufficient (which relates to your first question). 

To your second question, all of those (except for `repo_delete`) are associated with the `repoWriter` role which passes our `hasRepoWrite()` check. If we later allow a user to delete a mounted repo, that would be a slight tweak, which we can address if we decide to add that to notebooks.